core/src/main/java/com/rudderstack/android/sdk/core/RudderDeviceModeTransformationManager.java

```diff
@@ -104,37 +96,91 @@ public void run() {
                 , 1, 1, TimeUnit.SECONDS);
     }
 
-    private static String createDeviceTransformPayload(List<Integer> rowIds, List<String> messages) {
-        if (rowIds.isEmpty() || messages.isEmpty() || rowIds.size() != messages.size())
-            return null;
-        StringBuilder jsonPayload = new StringBuilder();
-        jsonPayload.append("{");
-        jsonPayload.append("\"batch\" :");
-        jsonPayload.append("[");
-        int totalBatchSize = Utils.getUTF8Length(jsonPayload) + 2;
-        try {
-            for (int i = 0; i < rowIds.size(); i++) {
-                StringBuilder message = new StringBuilder();
-                message.append("{");
-                message.append("\"orderNo\":").append(rowIds.get(i)).append(",");
-                message.append("\"event\":").append(messages.get(i));
-                message.append("}");
-                message.append(",");
-                totalBatchSize += Utils.getUTF8Length(message);
-                if (totalBatchSize >= Utils.MAX_BATCH_SIZE) {
-                    RudderLogger.logDebug(String.format(Locale.US, "DeviceModeTransformationManager: createDeviceTransformPayload: MAX_BATCH_SIZE reached at index: %d | Total: %d", i, totalBatchSize));
-                    break;
-                }
-                jsonPayload.append(message);
+    private void createMessageIdTransformationRequestMap() {
+        for (int i = 0; i < messageIds.size(); i++) {
+            RudderMessage message = gson.fromJson(messages.get(i), RudderMessage.class);
+            messageIdTransformationRequestMap.put(messageIds.get(i), message);
+        }
+    }
+
+    private boolean handleTransformationResponse(Result result, TransformationRequest transformationRequest) {
```

Comment:
Can we write a test case for this by mocking integration interface and checking if the data is valid?